pubsub/deployment/src/main/java/io/quarkiverse/googlecloudservices/pubsub/deployement/PubSubDevServiceProcessor.java

```diff
@@ -0,0 +1,168 @@
+        try {
+            devService = startContainerIfAvailable(dockerStatusBuildItem, pubSubBuildTimeConfig.devservice,
+                    globalDevServicesConfig.timeout);
+        } catch (Throwable t) {
+            LOGGER.warn("Unable to start PubSub dev service", t);
+            // Dump captured logs in case of an error
+            compressor.closeAndDumpCaptured();
+            return null;
+        } finally {
+            compressor.close();
+        }
+
+        return devService == null ? null : devService.toBuildItem();
+    }
+
+    /**
+     * Start the container if conditions are met.
+     *
+     * @param dockerStatusBuildItem, Docker status
+     * @param config, Configuration for the PubSub service
+     * @param timeout, Optional timeout for starting the service
+     * @return Running service item, or null if the service couldn't be started
+     */
+    private DevServicesResultBuildItem.RunningDevService startContainerIfAvailable(DockerStatusBuildItem dockerStatusBuildItem,
+            PubSubDevServiceConfig config,
+            Optional<Duration> timeout) {
+        if (!config.enabled) {
+            // PubSub service explicitly disabled
+            LOGGER.debug("Not starting Dev Services for PubSub as it has been disabled in the config");
+            return null;
+        }
+
+        if (!dockerStatusBuildItem.isDockerAvailable()) {
+            LOGGER.warn("Not starting devservice because docker is not available");
+            return null;
+        }
+
+        return startContainer(dockerStatusBuildItem, config, timeout);
+    }
+
+    /**
+     * Starts the PubSub emulator container with provided configuration.
+     *
+     * @param dockerStatusBuildItem, Docker status
+     * @param config, Configuration for the PubSub service
+     * @param timeout, Optional timeout for starting the service
+     * @return Running service item, or null if the service couldn't be started
+     */
+    private DevServicesResultBuildItem.RunningDevService startContainer(DockerStatusBuildItem dockerStatusBuildItem,
+            PubSubDevServiceConfig config,
+            Optional<Duration> timeout) {
+        // Create and configure PubSub emulator container
+        PubSubEmulatorContainer pubSubEmulatorContainer = new QuarkusPubSubContainer(
+                DockerImageName.parse(config.imageName).asCompatibleSubstituteFor("gcr.io/google.com/cloudsdktool/cloud-sdk"),
+                config.port.orElse(null));
+
+        // Set container startup timeout if provided
+        timeout.ifPresent(pubSubEmulatorContainer::withStartupTimeout);
+        pubSubEmulatorContainer.start();
+
+        // Return running service item with container details
+        return new DevServicesResultBuildItem.RunningDevService(PubSubBuildSteps.FEATURE,
+                pubSubEmulatorContainer.getContainerId(),
+                pubSubEmulatorContainer::close, "pubsub", pubSubEmulatorContainer.getEmulatorEndpoint());
```

Comment:
The `key` should be a config key, we don't have any for PubSub but it will be better to follow the naming and use something like `quarkus.google.cloud.pubsub.emulator-host`, it could then be used to init the `TransportChannelProvider`